packages/rule-stylesheet-limits/src/rule.ts

```diff
@@ -0,0 +1,164 @@
+                hasImportLimit = true;
+            }
+            if (options.maxRules && options.maxRules < maxRules) {
+                maxRules = options.maxRules;
+            }
+            if (options.maxSheets && options.maxSheets < maxSheets) {
+                maxSheets = options.maxSheets;
+            }
+        }
+
+        // The following function will be evaluated in the context of the page.
+        const injectedCode = function() {
+
+            // Recursively count rules and imports in the passed stylesheet.
+            const countRules = (styleSheet: CSSStyleSheet) => {
+
+                const results = {
+                    imports: 0,
+                    rules: 0,
+                    sheets: 1
+                };
+
+                try {
+
+                    // Count rules in this stylesheet.
+                    Array.from(styleSheet.cssRules).forEach((rule) => {
+
+                        if (rule instanceof CSSStyleRule) {
+
+                            // Count each selector in a style rule separately.
+                            results.rules += rule.selectorText.split(',').length;
+
+                        } else if (rule instanceof CSSImportRule) {
+
+                            // Recursively count rules in imported stylesheets.
+                            const subResults = countRules(rule.styleSheet);
+
+                            results.imports += Math.max(results.imports, subResults.imports + 1);
+                            results.rules += subResults.rules + 1;
+                            results.sheets += subResults.sheets;
+
+                        } else {
+
+                            // Other rules count as one each.
+                            results.rules += 1;
+
+                        }
+                    });
+
+                } catch (e) {
+
+                    /*
+                     * Accessing cssRules of a cross-origin stylesheet can throw.
+                     * If this happens, exclude the stylesheet from the count.
+                     */
+                }
+
+                return results;
+            };
+
+            const combinedResults = {
+                imports: 0,
+                rules: 0,
+                sheets: 0
+            };
+
+            // Get the recursive count of rules for all stylesheets in the page.
+            Array.from(document.styleSheets).forEach((sheet) => {
+                if (sheet instanceof CSSStyleSheet) {
+                    const subResults = countRules(sheet);
+
+                    combinedResults.imports += Math.max(combinedResults.imports, subResults.imports);
+                    combinedResults.rules += subResults.rules;
+                    combinedResults.sheets += subResults.sheets;
+                }
+            });
+
+            return combinedResults;
+        };
+
+        const validateScanEnd = async () => {
+
+            const results = await context.evaluate(`(${injectedCode})()`);
```

Comment:
Does these limits apply to other browsers or just to IE?

Reply:
Just IE. I'm sure other browsers have limits somewhere, but they haven't been well documented and online reports are inconsistent. The higher limits in IE10+ have been documented so I included them for completeness.

Reply:
Should we check then the target browsers before evaluate? If they are not targeting IE10, then the rules shouldn't be executed, right?

Reply:
Sorry, I meant just IE and Edge. I tested and confirmed the higher IE10 limits still apply to the latest Edge version.

Reply:
ok!